torch/_dynamo/variables/torch.py

```diff
@@ -382,8 +382,12 @@ def call_function(
             assert not (args or kwargs)
             install_guard(TorchFunctionDisableVariable._guards_singleton)
             return ConstantVariable.create(tx.output.torch_function_enabled)
+        elif self.value is torch.overrides.has_torch_function:
+            assert not (kwargs or len(args) != 1)
+            return ConstantVariable.create(
+                any(has_torch_function(a) for a in args[0].unpack_var_sequence(tx)),
+            )
```

Comment:
Can you merge it with the other place?
https://github.com/pytorch/pytorch/blob/da559c98e3d6a6b8672a0f923e76ae4b109ce4ac/torch/_dynamo/variables/torch.py#L398-L405

Reply:
We can probably merge it likes this. But IMO the original solution is stricter and cleaner. Please let me know your preference or if there's a better way to go
```
        elif self.value in (
            torch.overrides.has_torch_function,
            torch.overrides.has_torch_function_variadic,
            torch.overrides.has_torch_function_unary,
        ):
            assert not kwargs
            elems = (
                args[0].unpack_var_sequence(tx)
                if len(args) == 1 and isinstance(args[0], TupleVariable)
                else args
            )
            return ConstantVariable.create(
                any(has_torch_function(x) for x in elems),
            )
```

Reply:
If you want to update ```has_torch_function``` as this, it would cover more cases and more concise:
```
def has_torch_function(vt: "torch._dynamo.variables.base.VariableTracker", tx: "torch._dynamo.symbolic_convert.InstructionTranslatorBase") -> bool:
    from torch._dynamo.variables import UserDefinedObjectVariable
    from torch._dynamo.variables.torch_function import TensorWithTFOverrideVariable

    if vt.has_unpack_var_sequence(tx):
        return any(has_torch_function(v, tx) for v in vt.unpack_var_sequence(tx))
    else:
        return isinstance(vt, TensorWithTFOverrideVariable) or (
            isinstance(vt, UserDefinedObjectVariable)
            and hasattr(vt.value, "__torch_function__")
        )
```